tachyon/math/polynomial/domains/radix2/radix2_evaluation_domain_unittest.cc

```diff
@@ -0,0 +1,298 @@
+// Copyright 2022 arkworks contributors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.arkworks and the LICENCE-APACHE.arkworks
+// file.
+
+#include "tachyon/math/polynomial/domains/radix2/radix2_evaluation_domain.h"
+
+#include "gtest/gtest.h"
+
+#include "tachyon/math/elliptic_curves/bls/bls12_381/fr.h"
+#include "tachyon/math/polynomial/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::math {
+
+namespace {
+
+template <typename PrimeFieldType>
+class Radix2EvaluationDomainTest : public testing::Test {
+ public:
+  static void SetUpTestSuite() { PrimeFieldType::Init(); }
+};
+
+using PrimeFieldTypes = testing::Types<bls12_381::Fr>;
+
+TYPED_TEST_SUITE(Radix2EvaluationDomainTest, PrimeFieldTypes);
+
+TYPED_TEST(Radix2EvaluationDomainTest, VanishingPolynomialEvaluation) {
+  using F = TypeParam;
+
+  for (size_t coeffs = 1; coeffs < 10; ++coeffs) {
+    Radix2EvaluationDomain<F> domain(coeffs);
+    Radix2EvaluationDomain<F> coset_domain =
+        domain.GetCoset(F::FromMontgomery(F::Config::kSubgroupGenerator));
+    auto z = domain.GetVanishingPolynomial();
+    auto z_coset = coset_domain.GetVanishingPolynomial();
+    for (size_t i = 0; i < 100; ++i) {
+      F point = F::Random();
+      F res = z.Evaluate(point);
+      EXPECT_EQ(res, domain.EvaluateVanishingPolynomial(point));
+      EXPECT_EQ(z_coset.Evaluate(point),
+                coset_domain.EvaluateVanishingPolynomial(point));
+    }
+  }
+}
+
+TYPED_TEST(Radix2EvaluationDomainTest, VanishingPolynomialVanishesOnDomain) {
+  using F = TypeParam;
+
+  for (size_t coeffs = 0; coeffs < 10; ++coeffs) {
+    Radix2EvaluationDomain<F> domain(coeffs);
+    auto z = domain.GetVanishingPolynomial();
+    for (F& element : domain.GetElements()) {
+      EXPECT_TRUE(z.Evaluate(element).IsZero());
+    }
+
+    Radix2EvaluationDomain<F> coset_domain =
+        domain.GetCoset(F::FromMontgomery(F::Config::kSubgroupGenerator));
+    auto z_coset = coset_domain.GetVanishingPolynomial();
```

Comment:
Do you intend to use `auto`? Otherwise, please specify type. See https://google.github.io/styleguide/cppguide.html#Type_deduction